tgctoolbox/settings.py

```diff
@@ -0,0 +1,183 @@
+import logging
+import os
+from typing import List
+from typing import Optional
+
+import yaml
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class SettingsMeta(type):
+    """Metaclass to create a singleton Settings class."""
+
+    _instance = None
+
+    def __call__(cls, *args, **kwargs):
+        """Create a new instance of the class if it doesn't exist, otherwise return the existing instance.
+
+        Returns:
+            The existing instance of the class.
+        """
+        if cls._instance is None:
+            cls._instance = super().__call__(*args, **kwargs)
+        return cls._instance
+
+
+class Settings(metaclass=SettingsMeta):
+    """Class to manage application settings."""
+
+    def __init__(
+        self,
+        load_predefined: bool = True,
+        reload: bool = False,
+        load_env: bool = False,

```

Comment:
the use case of the loading var from env is also to overwrite the existing variablies in the .yaml files. No need of this flag, I would say that it is alwasy true.

We are loading the files in order:
base.yaml
local/prodcution.yaml
env

Reply:
Which flag exactly? Also, should I only load from other sources if some of the required settings are not loaded?

Reply:
No. If we want to load local, DO NOT load production and vice versa. load_env flag as it should be always true

Reply:
Hmm, sure, its gonna overwrite the yaml loaded settings in such a case, is `VAR_` prefix enough of a differentiation?

Reply:
Since [bb93b72](https://github.com/TwoGuysCoding/TGCToolbox/pull/5/commits/bb93b72da0b65f04af308b45ab4e4c1e6fca2992) env is always loaded.